packages/inquirer/lib/ui/baseUI.js

```diff
@@ -55,6 +55,10 @@ class UI {
     this.rl.output.end();
     this.rl.pause();
     this.rl.close();
+    // TODO: is there a better way
+    // of setting it to be called
+    this.rl.close.called = true;
+    this.rl.output.end.called = true;
```

Comment:
Could you explain why this was necessary?

Looks like there was some mock checks in the code; and that those stopped working. My hunch is that the mocks stopped working after the code change.

Reply:
```ReferenceError: module is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and '/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at file:///Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/lib/inquirer.js:22:18
    at ModuleJob.run (node:internal/modules/esm/module_job:198:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:409:24)
    at async importModuleDynamicallyWrapper (node:internal/vm/module:435:15)
    at async formattedImport (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/nodejs/esm-utils.js:7:14)
    at async exports.requireOrImport (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/nodejs/esm-utils.js:48:32)
    at async exports.loadFilesAsync (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/nodejs/esm-utils.js:103:20)
    at async singleRun (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/cli/run-helpers.js:125:3)
    at async exports.handler (/Users/unfocused/Documents/Projects/Inquirer.js/packages/inquirer/node_modules/mocha/lib/cli/run.js:374:5)```

Reply:
Since we converted to an ESModule, module does not exist in ES Module syntax

Reply:
> We are manually setting close & end called to true in BaseUI:

    // TODO: is there a better way
    // of setting it to be called
    this.rl.close.called = true;
    this.rl.output.end.called = true;
Please double check this & let us know if there is a proper solution for this. The test fails & we believe it is because NodeJS Readline is updated now due to using ESM.

Reply:
Please see my earlier comment about this. To summarize, I believe it might be due to the new NodeJS ReadLine API. Not sure where the called property is being added to the object (I think it is Mocha?) but it is not working for two test cases causing them to fail.

We did not alter the Mocha code & I saw the ReadLine API is different in the newer Node versions so perhaps it was added there so I made a hacky "fix".

I would like a proper fix so I'll see what our team can do :)

Reply:
Also fixed that, I guess it was caused by leaky test spies (like another test mocking some readline methods somehow polluting other tests...) I've fixed that by setting the correct spies on the test relying on them.